fem/pfespace.cpp

```diff
@@ -3507,6 +3507,125 @@ void ParFiniteElementSpace::Update(bool want_transform)
    }
 }
 
+void ParFiniteElementSpace::GetTrueUpdateOperator(OperatorHandle &T)
+{
+   bool need_P_old = false;
+   std::unique_ptr<HypreParMatrix> P_old;
+
+   T.Clear();  // preserves T.Type()
+
+   switch (mesh->GetLastOperation())
+   {
+      case Mesh::REFINE:
+      {
+         if (T.Type() == Operator::Hypre_ParCSR)
+         {
+            SetUpdateOperatorType(Operator::MFEM_SPARSEMAT);
+            need_P_old = true;
+         }
+         break;
+      }
+
+      case Mesh::DEREFINE:
+      {
+         if (T.Type() == Operator::Hypre_ParCSR)
+         {
+            need_P_old = true;
+         }
+         break;
+      }
+
+      case Mesh::REBALANCE:
+      {
+         if (T.Type() == Operator::Hypre_ParCSR)
+         {
+            need_P_old = true;
+         }
+         break;
+      }
+
+      default:
+         break;
+   }
+   if (need_P_old)
+   {
+      // Save P as P_old:
+      P_old.reset(Dof_TrueDof_Matrix()); // make sure P is constructed
```

Comment:
This is for the conforming case only, right? Should this be documented in the header file?

Reply:
`Dof_TrueDof_Matrix()` handles all cases (in the nonconforming case it is always constructed when constructing the FE space; in the conforming case it is constructed on demand) and the result is always a `HypreParMatrix` (this is what we want here), so there should be no restrictions.